app/views/search/find.html.erb

```diff
@@ -1,8 +1,15 @@
-<% @page_title = _("Search") -%>
+<% if @search_term == "" %>
+   <% @page_title = _('openSUSE Software Search') -%>
+<% else if @category != "" && @search_term == "" %>
+      <% @page_title = _("Category:") + " " + (@category.split /(?=[A-Z][a-z])/).join(" ") + " | " + _('openSUSE Software') -%>
+   <% else %>
+      <% @page_title = _("Searching for") + " " + @search_term + " | " + _('openSUSE Software') -%>
+   <% end %>
+<% end %>
```

Comment:
currently you do basically this:
```ruby
if @search_term == ""
  # first branch
else
  if @category != "" && @search_term == ""
    # second branch
  else
    # third branch
  end
end
```
That way you will never reach "second branch".  If `@search_term == ""` is true, you go directly to "first branch", it has to be false to even check `@category` which means  you have `@category != "" && false` which results in false.

You probably want to do this instead (using the rails methods for conditionals):
```ruby
if @category.present? && @search_term.blank?
  # second branch
elsif @search_term.blank?
  # first branch
else
  # third branch
end
```

Reply:
I have a third option! :wink: 

``` ruby
if @search_term.blank?
  if @category.present? 
    # second branch
  else
    # second branch
  end
else
  # third branch
end
```

@alexandergraul option is already fine, but I would ay that with this one we gain some more readability. :wink: